internal/provider/resource_docker_volume.go

```diff
@@ -129,6 +135,11 @@ func resourceDockerVolumeRead(ctx context.Context, d *schema.ResourceData, meta
 }
 
 func resourceDockerVolumeDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	if prevent_destroy := d.Get("prevent_destroy").(bool); prevent_destroy {
+		return nil
```

Comment:
```suggestion
		d.SetId("")
		return nil
```

https://learn.hashicorp.com/tutorials/terraform/provider-setup#implement-read

I don't have a confidence, but maybe `d.SetId("")` is required.

Otherwise after you remove the resource definition,
`terraform plan` would include the destroy operation because the resource remains in the State.

Reply:
I added `d.SetId("")` and got the same result. This works the same was as `keep_locally` for docker_images in the provider.

Reply:
Oh, probably I misunderstood.
`d.SetId("")` isn't needed.